doc/api/package.json.md

```diff
@@ -0,0 +1,345 @@
+{
+  "main": "./main.js",
+  "exports": {
+    "browser": "./feature-browser.mjs",
+    "node": {
+      "import": "./feature-node.mjs",
+      "require": "./feature-node.cjs"
+    }
+  }
+}
+```
+
+Conditions continue to be matched in order as with flat conditions. If
+a nested conditional does not have any mapping it will continue checking
+the remaining conditions of the parent condition. In this way nested
+conditions behave analogously to nested JavaScript `if` statements.
+
+### `"main"`
+<!-- YAML
+added: v0.4.0
+-->
+
+```json
+{
+    "main": "./main.js"
+}
+```
+
+The `"main"` field defines the script that is used when the
+current directory is required by another script.
+
+```js
+require('./path/to/directory'); // This resolves to ./path/to/directory/main.js.
+```
+
+This field is ignored when the [`"exports"`][] field is provided. The `"main"`
+field is supported in all versions of Node.js, but its capabilities are limited:
+it only defines the main entry point of the package.
+
+### `"type"`
+<!-- YAML
+added: v12.0.0
+changes:
+  - version: v13.2.0
+    pr-url: https://github.com/nodejs/node/pull/29866
+    description: Unflag --experimental-modules.
+-->
+
+The `"type"` field defines how `.js` files should be treated
+within a particular `package.json` file’s package scope.
+
+Files ending with `.js` will be loaded as ES modules when the nearest parent
+`package.json` file contains a top-level field `"type"` with a value of
+`"module"`.
+
+The nearest parent `package.json` is defined as the first `package.json` found
+when searching in the current folder, that folder’s parent, and so on up
+until the root of the volume is reached.
+
+```json
+// package.json
+{
+  "type": "module"
+}
+```
+
+```sh
+# In same folder as above package.json
+node my-app.js # Runs as ES module
+```
+
+If the nearest parent `package.json` lacks a `"type"` field, or contains
+`"type": "commonjs"`, `.js` files are treated as CommonJS. If the volume root is
+reached and no `package.json` is found, Node.js defers to the default, a
+`package.json` with no `"type"` field.
+
+`import` statements of `.js` files are treated as ES modules if the nearest
+parent `package.json` contains `"type": "module"`.
+
+<!-- eslint-skip -->
+```js
+// my-app.js, part of the same example as above
+import './startup.js'; // Loaded as ES module because of package.json
+```
+
+Package authors should include the `"type"` field, even in packages where all
+sources are CommonJS. Being explicit about the `type` of the package will
+future-proof the package in case the default type of Node.js ever changes, and
+it will also make things easier for build tools and loaders to determine how the
+files in the package should be interpreted.
+
+Regardless of the value of the `"type"` field, `.mjs` files are always treated
+as ES modules and `.cjs` files are always treated as CommonJS.
+
+If the `"type"` field is not provided or is using an unsupported value, it is
+treated as `"commonjs"`.
```

Comment:
Perhaps we can remove this line, and then just include the `"or is using an unsupported value"` in one of the descriptions of this above, since there is quite a bit of repetition here.